File: SkaiLab.Invoice.Models/Expense.cs
using System;
using System.Collections.Generic;

namespace SkaiLab.Invoice.Models
{
    public class Expense
    {
        public long Id { get; set; }
        public long VendorId { get; set; }
        public string OrganisationId { get; set; }
        public Vendor Vendor { get; set; }
        public DateTime Created { get; set; }
        public DateTime Date { get; set; }
        public DateTime? DeliveryDate { get; set; }
        public string OrderNumber { get; set; }
        public string Note { get; set; }
        public int CurrencyId { get; set; }
        public Currency Currency { get; set; }
        public decimal TaxCurrencyExchangeRate { get; set; }
        public decimal BaseCurrencyExchangeRate { get; set; }
        public decimal Total { get; set; }
        public decimal TotalIncludeTax { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? ApprovedDate { get; set; }
        public string ApprovedBy { get; set; }
        public DateTime? BilledDate { get; set; }
        public string BilledBy { get; set; }
        public string CustomerInvoiceUrl { get; set; }
        public int ExpenseStatusId { get; set; }
        public ExpenseStatus ExpenseStatus { get; set; }
        public string RefNo { get; set; }
        public string DeletedBy { get; set; }
        public DateTime? DeletedDate { get; set; }
        public List<Attachment> Attachments { get; set; }
        public List<PurchaseOrderItem> ExpenseItems { get; set; }
        public Organsation Organisation { get; set; }
        public string TermAndCondition { get; set; }
        public DateTime? CloseDate { get; set; }
        public bool HasCloseDoc { get; set; }
        public Attachment CloseAttachment { get; set; }
    }
    public class ExpenseForUpdate
    {
        public Expense Expense { get; set; }
        public List<Currency> Currencies { get; set; }
        public List<Location> Locations { get; set; }
        public List<Tax> Taxes { get; set; }
        public int BaseCurrencyId { get; set; }
        public Currency TaxCurrency { get; set; }

    }
    public class ExpenseStatus
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public long Count { get; set; }
    }
    public class ExpenseItem
    {
        public long Id { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public double? DiscountRate { get; set; }
        public long? TaxId { get; set; }
        public Tax Tax { get; set; }
        public decimal LineTotal { get; set; }
        public decimal LineTotalIncludeTax { get; set; }
        public string Description { get; set; }
        public long ExpenseId { get; set; }
    }
}

File: SkaiLab.Invoice/ClientApp/src/app/models/expense.ts
import { Attachment } from "./attachment";
import { Currency } from "./currency";
import { ExepnseStatusEnum } from "./enum";
import { PurchaseOrderItem } from "./purchaseOrder";
import { Tax } from "./tax";
import { Vendor } from "./vendor";

export class ExpenseStatus{
    id:number;
    name:string;
    count:number;
    constructor(){
        this.id=0;
        this.name="";
        this.count=0;
    }
}
export class Expense{
    id:number;
    vendorId:number;
    organisationId:string;
    vendor:Vendor;
    creaed:Date;
    date:Date;
    deliveryDate:Date;
    orderNumber:string;
    note:string;
    currencyId:number;
    currency:Currency;
    taxCurrencyExchangeRate:number;
    baseCurrencyExchangeRate:number;
    total:number;
    totalIncludeTax:number;
    createdBy:string;
    approvedDate:Date;
    approvedBy:string;
    billedDate:Date;
    billedBy:string;
    customerInvoiceUrl:string;
    expenseStatusId:number;
    expenseStatus:ExpenseStatus;
    refNo:string;
    selected:boolean;
    attachments:Attachment[];
    expenseItems:PurchaseOrderItem[];
    termAndCondition:string;
    closeDate?:Date;
    hasCloseDoc:boolean;
    loading:boolean;
    closeAttachment:Attachment;
    constructor(){
        this.id=0;
        this.vendorId=null;
        this.organisationId="";
        this.vendor=new Vendor();
        this.creaed=new Date();
        this.date=new Date();
        this.deliveryDate=new Date();
        this.orderNumber="";
        this.note="";
        this.currencyId=null;
        this.currency=new Currency();
        this.taxCurrencyExchangeRate=0;
        this.baseCurrencyExchangeRate=0;
        this.total=0;
        this.totalIncludeTax=0;
        this.createdBy="";
        this.approvedDate=null;
        this.approvedBy="";
        this.billedDate=null;
        this.billedBy="";
        this.customerInvoiceUrl="";
        this.expenseStatusId=ExepnseStatusEnum.Draft;
        this.expenseStatus=new ExpenseStatus();
        this.selected=false;
        this.attachments=[];
        this.expenseItems=[];
        this.termAndCondition="";
        this.closeDate=null;
        this.hasCloseDoc=false;
        this.closeAttachment=new Attachment();
    }
}
export class ExpenseForUpdate{
    expense:Expense;
    currencies:Currency[];
    taxes:Tax[];
    baseCurrencyId:number;
    taxCurrency:Currency;
    constructor(){
        this.expense=new Expense();
        this.currencies=[];
        this.taxes=[];
        this.baseCurrencyId=0;
        this.taxCurrency=new Currency();
    }
}